src/components/form/form.tsx

```diff
@@ -1,68 +1,107 @@
-import { Button, Field, FormGeneric, FormPanel } from '@mdb/flora';
+import { Button, Field, FormGeneric, FormPanel, FormValues } from '@mdb/flora';
+import countryList from 'react-select-country-list';
 import {
+  Registration,
   jobFunctions,
   institutionTypes,
   teachingStatuses,
-} from 'src/lib/registration';
+} from 'lib/registration';
+import { uploadToAppServices } from 'lib/app-services';
 import FormProps from './types';
 import styles from './styles';
 
+export function isRequired() {
+  return function (value: string): string {
+    return !value || value === '' ? 'This field is required' : '';
+  };
+}
+
+export function emailPattern() {
+  return function (value: string): string {
+    if (value) {
+      return value.match(/^\S+@\S+\.\S+$/)
+        ? ''
+        : 'Please enter a valid email address';
+    }
+    return '';
+  };
+}
+
 interface FieldInterface {
   name: string;
   label: string;
   options?: Array<string>;
+  type?: 'text' | 'email';
   component: 'text-input' | 'select' | 'text-area' | 'checkbox';
+  validators?: Array<{
+    rule?: string;
+    (value: string): string;
+  }>;
 }
 
 const fields: Array<FieldInterface> = [
   {
     name: 'firstName',
     label: 'First Name',
+    type: 'text',
     component: 'text-input',
+    validators: [isRequired()],
   },
   {
     name: 'lastName',
     label: 'Last Name',
+    type: 'text',
     component: 'text-input',
+    validators: [isRequired()],
   },
   {
     name: 'email',
     label: 'Email',
+    type: 'email',
     component: 'text-input',
+    validators: [isRequired(), emailPattern()],
   },
   {
     name: 'jobFunction',
     label: 'Your Job Function',
     component: 'select',
     options: jobFunctions,
+    validators: [isRequired()],
   },
```

Comment:
So we still need to account for the case where they select "Other". We need to have another text field appear so they can put in something.

Reply:
for sure, it's on my agenda for UX review with Liz tomorrow so was going to handle that once I get a clear answer